crates/task-impls/src/network.rs

```diff
@@ -248,6 +248,14 @@ impl<
     }
 
     async fn cancel_subtasks(&mut self) {}
+
+    async fn periodic_task(&self, task_id: String, sender: &Sender<Arc<Self::Event>>) {
```

Comment:
Why don't we just define these in the trait? Looks like all instances of this method do the same thing?

Reply:
only health check task behaves differently, as that will have logic for the logging if heartbeat missed for some time. i can make a trait and have this as default implementation

Reply:
let me know what you think about this new approach